reactive/kubernetes_master.py

```diff
@@ -3546,3 +3581,71 @@ def send_default_cni():
 @when("config.changed.default-cni")
 def default_cni_changed():
     remove_state("kubernetes-master.components.started")
+
+
+@when("kubernetes-master.components.started", "kubernetes-master.apiserver.configured")
+@when_not("kubernetes-master.kubelet.configured")
+def configure_kubelet():
+    uid = hookenv.local_unit()
+    username = "system:node:{}".format(get_node_name().lower())
+    group = "system:nodes"
+    token = get_token(username)
+    if not token:
+        setup_tokens(None, username, uid, group)
+        token = get_token(username)
+    if not token:
+        hookenv.log(
+            "Failed to create token for {}; will retry".format(username),
+            hookenv.WARNING,
+        )
+        return
+
+    local_endpoint = kubernetes_master.get_local_api_endpoint()
+    local_url = kubernetes_master.get_api_url(local_endpoint)
+    create_kubeconfig(
+        kubelet_kubeconfig_path, local_url, ca_crt_path, token=token, user="kubelet"
+    )
+
+    dns_ready, dns_ip, dns_port, dns_domain = get_dns_info()
+    if not dns_ready:
+        hookenv.log("DNS not ready, waiting to configure Kubelet")
+        return
+    dns_info = [dns_ip, dns_port, dns_domain]
+    db.set("kubernetes-master.kubelet.dns-used", dns_info)
+
+    registry = hookenv.config("image-registry")
+    taints = hookenv.config("register-with-taints").split()
+    kubernetes_common.configure_kubelet(dns_domain, dns_ip, registry, taints=taints)
+    service_restart("snap.kubelet.daemon")
+
+    set_flag("kubernetes-master.kubelet.configured")
+
+
+@when_any("config.changed.kubelet-extra-args", "config.changed.kubelet-extra-config")
```

Comment:
Are you planning another PR with `kubelet-extra-args` or `kubelet-extra-config` in the `config.yaml`?

Reply:
i think the `layer-kubernetes-common` looks up that config.  This is probably the right time to pull those config items from `kubernetes-worker`

Reply:
Yup, a PR for this already exists, but it is targeted against a base layer that's shared by both kubernetes-master and kubernetes-worker: https://github.com/charmed-kubernetes/layer-kubernetes-master-worker-base/pull/13

Reply:
🤦🏻‍♂️ oh there they are!  All's well